[IMP] Taxes: improve tax configuration doc (#14908)
odoo/documentation: content/applications/finance/accounting/taxes.rst

Tax computation
~~~~~~~~~~~~~~~

The :guilabel:`Tax Computation` field determines how the tax amount is computed from the sales
price. The following options are available:

- :ref:`taxes/computation/group-of-taxes`: a combination of several other taxes
- :ref:`taxes/computation/fixed`: a fixed amount
- :ref:`taxes/computation/percentage-of-price`: a percentage of the tax-excluded sales price
- :ref:`taxes/computation/percentage-of-price-tax-included`: a percentage of the tax-included total
- :ref:`taxes/computation/python-code`: a custom user-defined formula.

.. _taxes/computation/group-of-taxes:

Group of Taxes
**************

  The tax is a combination of multiple sub-taxes. You can add as many taxes as you want, in the
  order you want them to be applied.

.. important::
   Make sure that the tax sequence is correct, as the order in which they are may impact the

>    taxes' amounts computation, especially if one of the taxes :ref:`affects the base of the

Reviewer: ```suggestion
   taxes' amounts computation, especially if any of the taxes :ref:`affect the base of the
```

Tax computation
~~~~~~~~~~~~~~~

The :guilabel:`Tax Computation` field determines how the tax amount is computed from the sales
price. The following options are available:

- :ref:`taxes/computation/group-of-taxes`: a combination of several other taxes
- :ref:`taxes/computation/fixed`: a fixed amount
- :ref:`taxes/computation/percentage-of-price`: a percentage of the tax-excluded sales price
- :ref:`taxes/computation/percentage-of-price-tax-included`: a percentage of the tax-included total
- :ref:`taxes/computation/python-code`: a custom user-defined formula.

.. _taxes/computation/group-of-taxes:

> Group of Taxes

Reviewer 2: ```suggestion
Group of taxes
```
We use [sentence case](https://www.odoo.com/documentation/18.0/contributing/documentation/content_guidelines.html#document-structure) for headings, even though this contradicts another [guideline](https://www.odoo.com/documentation/18.0/contributing/documentation/content_guidelines.html#capitalization), which is to refer to Odoo's UI elements as they appear. In any case, sentence case takes precedence :)
You will need to adapt the references above (e.g., ```:ref:`Group of Taxes <taxes/computation/group-of-taxes>` ``` to follow the second guidelines :)

Author: This makes sense, thank you very much!

>   The tax has a fixed amount in the default currency. The amount remains the same, regardless of the

Reviewer: I know you weren't the first one to decide to use bullet points here, but let's make these H5s instead since it's awkward to have this really spaced out bulleted list. Let's do this:
**Tax computation H4**
Sentence introducing the concept of what this field does.

- Bulleted list of references to the H5s, one for each of the tax computation options

H5s for each of the tax computation options (the content of these can stay the same as they are currently)

  sales price.

  .. example::

>      A product has a sales price of $1000, and we apply a $10 :guilabel:`fixed` tax. We then have:

Reviewer: Since we're just talking about the concept of a fixed tax here and not saying "On the tax record, set the :guilabel:`Tax Computation` field to :guilabel:`Fixed`", we should not use the guilabel.

        \text{sales price} \times \text{tax rate}`.

        .. example::
           A product has a sales price of $1000, and we apply a 10%
           :guilabel:`Percentage of Price Tax Included` tax that is :guilabel:`Included in Price`.
           We then have:

           +-------------+-------------+----------+----------+
           | Product     | Price       | Tax      | Total    |
           | sales price | without tax |          |          |
           +=============+=============+==========+==========+
           | 1,000       | 900         | 100      | 1,000.00 |
           +-------------+-------------+----------+----------+

           Note that the real tax rate in terms of the tax-excluded price is
           :math:`\frac{100}{900} = 11.111\%`.

  .. warning::
     This tax computation is rarely used and only useful in countries (e.g. Brazil, Bolivia) that
     quote tax rates as a percentage of the tax-included total.
     For the more usual need to compute tax amounts from a tax-included price, use the
     :guilabel:`Percentage of Price` tax computation with the :ref:`taxes/included-in-price`
     option.

> - **Python code**

Reviewer: Note this has changed to Custom Formula as of 18. We can handle that in a fw port :) make sure to change it in this whole section and not just the heading!

   | 1,000       | 1,000       | 10       | 1,010.00 |
   +-------------+-------------+----------+----------+

.. _taxes/computation/percentage-of-price:

Percentage of Price
*******************

The tax rate is a percentage of the **tax-excluded** subtotal.

> The exact tax computation depends on the :ref:`taxes/included-in-price` field, which determines

Reviewer 2: ```suggestion
The exact tax computation depends on the :ref:`Included in Price <taxes/included-in-price>` field, which determines
```
To use the field's label as on the UI.

         +=============+=============+==========+==========+
         | 1,000       | 1,000       | 111.11   | 1,111.11 |
         +-------------+-------------+----------+----------+

         Note that the real tax rate in terms of the tax-excluded price is
         :math:`\frac{111.11}{1000} = 11.111\%`.

   .. tab:: Tax-included

      If :guilabel:`Included in Price` is enabled, the computation is :math:`\text{tax amount} =
      \text{sales price} \times \text{tax rate}`.

      .. example::
         A product has a sales price of $1000, and we apply a 10%
         :guilabel:`Percentage of Price Tax Included` tax that is :guilabel:`Included in Price`.
         We then have:

         +-------------+-------------+----------+----------+
         | Product     | Price       | Tax      | Total    |
         | sales price | without tax |          |          |
         +=============+=============+==========+==========+
         | 1,000       | 900         | 100      | 1,000.00 |
         +-------------+-------------+----------+----------+

         Note that the real tax rate in terms of the tax-excluded price is
         :math:`\frac{100}{900} = 11.111\%`.

> .. warning::

Author: I've done that - thanks for the feedback on using `important` vs `warning`.

  :guilabel:`Python Code` defines the amount of the tax, and :guilabel:`Applicable Code` defines if

>   the tax is to be applied. The formula is found at the bottom of the :guilabel:`Definition` tab.

Reviewer: ```suggestion
  the tax is to be applied. Enter a formula for each field at the bottom of the :guilabel:`Definition` tab.
```

Included in price
~~~~~~~~~~~~~~~~~

With this option activated, the tax will treat the sales price on which it is applied as a total
including the tax amount. The tax computation will split the sales price into a base amount and a
tax amount. This makes it suitable for B2C sales in most countries where prices are quoted
tax-inclusive.

`Total = Sales Price = Computed Tax-Excluded price + Tax`

.. example::

>    A product has a sales price of $1000, and we apply a 10% :guilabel:`Percentage of Price` tax

Reviewer: The guilabel makes more sense to me here than it did with the fixed tax example. Let's talk in person about which is a better fit :)

      by preceding taxes` will "see" a modified sales price equal to the tax-excluded price plus
      the tax amount.

      .. example::
         A product has a sales price of $1000, and we apply a 10% :guilabel:`Percentage of Price`
         tax with :guilabel:`Affect base of subsequent taxes`. Any subsequent tax with
         :guilabel:`Base affected by preceding taxes` will "see" a modified sales price of $1100.

   .. tab:: Tax-included
      If :guilabel:`Included in Price` is checked, subsequent taxes with :guilabel:`Base affected
      by preceding taxes` will "see" a modified sales price equal to the tax-included price minus
      the tax amount.

      .. example::
         A product has a sales price of $1100, and we apply a 10% :guilabel:`Percentage of Price`
         tax with :guilabel:`Included in Price` and :guilabel:`Affect base of subsequent taxes`.
         Any subsequent tax with :guilabel:`Base affected by preceding taxes` will "see" a modified
         sales price of $1000.

You can configure a new :ref:`group of taxes <taxes/computation>` to include this tax or add it

> directly to a product line.

Reviewer: ```suggestion
This setting is considered any time multiple taxes are applied to the same product line, whether via a :ref:`group of taxes <taxes/computation>` or multiple taxes added directly to a product line.
```
I know this wasn't your text, but this change makes it easier to understand why we're mentioning this information

   .. image:: taxes/list-sequence.png
      :alt: The taxes' sequence in Odoo determines which tax is applied first

.. _taxes/base-affected:

> Base affected by preceding taxes

Reviewer: Let's mention that this setting is only visible when using the developer mode and link to [this page](https://www.odoo.com/documentation/master/applications/general/developer_mode.html)

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

This setting determines whether any previous tax with :ref:`taxes/base-subsequent` will modify the

> sales price "seen" by this tax.

Reviewer: ```suggestion
sales price that this tax is based on.
```
Or, if you prefer, "sales price that this tax uses." Either works for me :)
